pepper/apps/hash_transform.c

```diff
@@ -1,41 +1,30 @@
 #include "hash_transform.h"
 #include "dex_common.h"
+#include "hashing.h"
 
 struct ShaHash { bool digest[256]; };
-struct PedersenHash { uint32_t values[2]; }; /* [x,y] */
 
 void compute(struct In *input, struct Out *output){
     // Read private input
     struct Private pInput = readPrivateInput();
     struct ShaHash shaHash[1] = { 0 };
-    struct PedersenHash pedersenHash[1] = { 0 };
 
     bool shaIn[512] = { 0 };
-    bool pedersenInput[508] = { 0 };
-
     uint32_t index;
-    for (index=0; index < ORDERS; index++) {
+    for (index=0; index < ORDERS*253; index+=253) {
         // sha hash
-        copyBits(shaHash->digest, shaIn, 256);
-        copyBits(pInput.orders[index], shaIn+259, 253);
+        copyBits(shaHash->digest, 0, shaIn, 0, 256);
+        copyBits(pInput.orders, index, shaIn, 259, 253);
         ext_gadget(shaIn, shaHash, 0);
+        uint32_t result = sumBits(shaHash->digest, 256);
```

Comment:
this result assignment is not needed, right?

Reply:
good catch, removed it